### DISCUSSION TITLE
How to move wraped text second line forward in terminal and keep style?

### DISCUSSION THREAD

**Discussion by AntsForBreakfast** [Textualize/rich#3475] [General] [answered]
How to move wraped text second line forward in terminal and keep style?

**Comment by TomJGooding** [comment]
Could you provide a simple example to explain what you mean?

**Reply by AntsForBreakfast** [reply]
text = 'There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.'

I want it to be:
  1  There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form,
  |   by injected humour, or randomised words which don't look even slightly believable.'
  
  basically wrap just wraps into new line but i want to put something in front at the each wraped line

**Reply by TomJGooding** [reply, marked as answer]
Sorry I'm afraid I still don't understand what you are asking, but maybe something like this?

```python
from rich.console import Console
from rich.text import Text

TEXT = "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable."

console = Console()

rich_text = Text(TEXT)
wrapped_lines = rich_text.wrap(
    console,
    width=120,  # Replace this with the desired width or the console.width
)

console.print(wrapped_lines[0])
for line in wrapped_lines[1:]:
    console.print("|", line)
```
